### DISCUSSION TITLE
Logging changes, and the pathname for local versus deployed

### DISCUSSION THREAD

**Discussion by JimNayzium** [sveltejs/kit#10913] [General]
I am successfully writing a log.txt file with all the information I need on my local npm run dev version using this:

```
const logFilePath = `${self.rootPath}/src/lib/logs/log.txt`; // replace with the path to your log file
const logMessage = `${new Date().toISOString()} - ${message}\n`;
await fs.appendFile(logFilePath, logMessage);

```
My question is after I do a full build and upload to my hosted environment, will this path hold true? It feels like /src won't be in the final build path? 

What is the best way to go about coding this type of solution that would work locally and after deployment?

I tried using local path names (i.e. like ../../path_here) but it would not work in the dev environment that way for some reason.

I come from a PHP background and a novice one at that, so I am used to using $_SERVER['DOCUMENT_ROOT'] whenever I need to do something similarly.

Thanks!

**Comment by teemingc** [comment]
Hey, I’m not too sure myself but maybe you can try the below for getting a consistent path in node:

process.cwd()
https://www.geeksforgeeks.org/node-js-process-cwd-method/amp/
